File: app/src/main/java/com/example/android_challenge_btg/MainActivity.java
package com.example.android_challenge_btg;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;


import com.example.android_challenge_btg.fragments.FavoritesFragment;
import com.example.android_challenge_btg.fragments.MoviesFragment;
import com.google.android.material.tabs.TabLayout;

import butterknife.BindView;
import butterknife.ButterKnife;

public class MainActivity extends AppCompatActivity {

    @BindView(R.id.tabCategories)
    TabLayout tabLayout;
    @BindView(R.id.viewPager)
    ViewPager viewPager;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        setupViewPager();
    }

    private void setupViewPager(){
        ViewPagerAdapter adapter = new ViewPagerAdapter(getSupportFragmentManager());
        adapter.addFragment(new MoviesFragment(), "Movies");
        adapter.addFragment(new FavoritesFragment(), "Favorites");
        viewPager.setAdapter(adapter);
        tabLayout.setupWithViewPager(viewPager);
    }
}

File: app/src/main/java/com/arturkida/popularmovies_kotlin/MainActivity.kt
package com.arturkida.popularmovies_kotlin

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import com.arturkida.popularmovies_kotlin.data.ApiImpl

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val api = ApiImpl()

        api.getGenres()
    }
}